isic_kredit/ViewControllers/CreditViewController.swift

```diff
@@ -0,0 +1,255 @@
+            let settingsController = SettingsViewController()
+            navigationController?.pushViewController(settingsController, animated: true)
+        } else {
+            self.reloadData()
+        }
+        
+        
+    }
+    
+    override func loadView() {
+        super.loadView()
+        view.backgroundColor = ColorConstants.BackgroundColor
+        
+        let reloadButton = UIButton()
+        reloadButton.setImage(UIImage(named: "reloadIcon"), for: .normal)
+        self.reloadButton = reloadButton
+        
+        let accountButton = UIButton()
+        accountButton.setImage(UIImage(named: "accountIcon"), for: .normal)
+        self.accountButton = accountButton
+        
+        let buttonsStack = UIStackView(arrangedSubviews: [reloadButton, accountButton])
+        buttonsStack.spacing = 30
+        buttonsStack.distribution = .fillEqually
+        
+        let textLabel = UILabel()
+        textLabel.text = NSLocalizedString("Account balance", comment: "")
+        textLabel.textColor = ColorConstants.LabelsColor
+        textLabel.font = FontConstants.LabelsFont
+        
+        let moneyLabel = UILabel()
+        moneyLabel.text = "0 \(NSLocalizedString("CZK", comment: ""))"
+        moneyLabel.textColor = .white
+        moneyLabel.font = FontConstants.MoneyFont
+        moneyLabel.adjustsFontForContentSizeCategory = true
+        self.moneyLabel = moneyLabel
+    
+        let mainStack = UIStackView(arrangedSubviews: [textLabel, moneyLabel, buttonsStack])
+        mainStack.spacing = 30
+        mainStack.distribution = .fillEqually
+        mainStack.axis = .vertical
+        mainStack.alignment = .center
+        view.addSubview(mainStack)
+        mainStack.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+            make.leading.equalTo(view.safeAreaLayoutGuide)
+        }
+        
+    }
+    
+    /*
+     When account button is tapped show settings view controller
+     */
+    @objc func accountButtonTapped(_ sender: UIButton) {
+        
+        let settingsController = SettingsViewController()
+        navigationController?.pushViewController(settingsController, animated: true)
+    }
+    
+    /*
+     When reload button is tapped reload data from website
+     */
+    @objc func reloadButtonTapped(_ sender: UIButton) {
+        self.reloadData()
+    }
+    
+    /*
+     This function log in to https://agata.suz.cvut.cz/secure/index.php where is information about users money.
+     If there is some error function shows alert.
+     */
+    func reloadData() {
+        
+        self.hud.indicatorView = JGProgressHUDIndeterminateIndicatorView()
+        self.hud.textLabel.text = NSLocalizedString("Loading", comment: "")
+        self.hud.show(in: self.view)
+        
+        //Request website
+        Alamofire.request("https://agata.suz.cvut.cz/secure/index.php").responseString { agataResponse in
```

Comment:
Tenhle cell pekelnej chain patří do zvláštní třídy - nějaká api vrstva. Stejně tak by to mělo být rozdělené na nějaké jednotlivé funkce, v tomhle se nikdo nikdy nevyzná 😞

Reply:
Trida DataManager, chain je rozdelenej do nekolika funkci